pkg/pillar/cmd/volumemgr/blob.go

```diff
@@ -0,0 +1,301 @@
+			changed = true
+		}
+	}
+	if blob.WaitingForCerts {
+		log.Infof("Waiting for certs for blob %s", blob.Sha256)
+		return changed
+	}
+	log.Infof("Waiting for download for blob %s", blob.Sha256)
+	return changed
+}
+
+// verifyBlob ensure a particular blob is being verified
+// Updates status with WaitingForCerts
+func verifyBlob(ctx *volumemgrContext, objType string, sv SignatureVerifier, blob *types.BlobStatus) bool {
+	changed := false
+	if !blob.HasVerifierRef {
+		if blob.State == types.DOWNLOADED {
+			blob.State = types.VERIFYING
+			changed = true
+		}
+		done, errorAndTime := MaybeAddVerifyImageConfigBlob(ctx, objType, *blob, sv)
+		if done {
+			blob.HasVerifierRef = true
+			changed = true
+			return changed
+		}
+		// if errors, set the certError flag
+		// otherwise, mark as waiting for certs
+		if errorAndTime.HasError() {
+			blob.SetError(errorAndTime.Error, errorAndTime.ErrorTime)
+			changed = true
+		} else if !blob.WaitingForCerts {
+			blob.WaitingForCerts = true
+			changed = true
+		}
+	}
+	return changed
+}
+
+// lookForVerifiedBlob handles the split between PersistImageStatus and
+// VerifyImageStatus. If it only finds the Persist it returns nil but
+// sets up a VerifyImageConfig.
+// Also returns changed=true if the VolumeStatus is changed
+func lookForVerifiedBlob(ctx *volumemgrContext, objType string, sv SignatureVerifier, blob *types.BlobStatus) *types.VerifyImageStatus {
+	vs := lookupVerifyImageStatus(ctx, objType, blob.Sha256)
+	if vs == nil {
+		ps := lookupPersistImageStatus(ctx, objType, blob.Sha256)
+		if ps == nil || ps.Expired {
+			log.Infof("Verify/PersistImageStatus for blob %s not found", blob.Sha256)
+		} else {
+			log.Infof("lookForVerifiedBlob: Found PersistImageStatus for blob %s", blob.Sha256)
+			if !blob.HasPersistRef {
+				log.Infof("lookForVerifiedBlob: Adding PersistImageStatus reference for BlobStatus: %s", blob.Sha256)
+				AddOrRefCountPersistImageStatus(ctx, ps.Name, ps.ObjType, ps.FileLocation, ps.ImageSha256, ps.Size)
+				blob.HasPersistRef = true
+			}
+
+			// mark the BlobStatus state as VERIFIED as we already have a PersistImageStatus for the blob
+			if blob.State != types.VERIFIED {
+				blob.State = types.VERIFIED
+			}
+			if blob.Path != ps.FileLocation {
+				blob.Path = ps.FileLocation
+				log.Infof("lookForVerifiedBlob: Update FileLocation for %s: %s", blob.Sha256, blob.Path)
+			}
+			// If we don't already have a RefCount add one
+			if !blob.HasVerifierRef {
+				log.Infof("!HasVerifierRef")
+				// We don't need certs since Status already exists
+				MaybeAddVerifyImageConfigBlob(ctx, objType, *blob, sv)
+				blob.HasVerifierRef = true
+			}
+			//Wait for VerifyImageStatus to appear
+			return nil
+		}
+	} else {
+		log.Infof("Found verified blob status based on sha %s", blob.Sha256)
+		// If we don't already have a RefCount add one
+		// No need to checkCerts since we have a VerifyImageStatus
+		verifyBlob(ctx, objType, sv, blob)
```

Comment:
Here you need to update state based on VerifyImageStatus; verifyBlob doesn't do that AFAICT

Reply:
Correct. `verifyBlob` checks for `HasVerifierRef`:

* has ref? do nothing
* doesn't have ref? add `VerifyImageConfig{}` to start the verifier

> update state based on VerifyImageStatus

It does. This is just checking the `VerifyStatus` and retrieving it. It is updated [here](https://github.com/lf-edge/eve/pull/1110/files#diff-eb3ec09d4c0f3b985be7625eded82029R106) for `ContentTree` and [here](https://github.com/lf-edge/eve/pull/1110/files#diff-f8d21b734fa414774d21d4219a2676b1R58) for `OldVolumeStatus`.